libs/sdk-ui-pivot/src/impl/data/dataSource.ts

```diff
@@ -156,13 +157,23 @@ export class AgGridDatasource implements IDatasource {
                 dimensionSetTotals(definition.dimensions[1], desiredRowTotals),
             );
 
+        const columnBucket = bucketsFind(definition.buckets, BucketNames.COLUMNS);
+        const attributeBucket = bucketsFind(definition.buckets, BucketNames.ATTRIBUTE);
+
         // Update buckets property only when adding or removing columns totals/subtotals. So table is updated properly.
-        if (desiredRowTotals.length > 0 || (desiredRowTotals && definition.buckets[2].totals !== undefined)) {
-            transformedExecution = transformedExecution.withBuckets(
-                definition.buckets[0],
-                bucketSetTotals(definition.buckets[1], desiredTotals),
-                bucketSetTotals(definition.buckets[2], desiredRowTotals),
-            );
+        if (desiredRowTotals.length > 0 || (desiredRowTotals && columnBucket?.totals !== undefined)) {
+            if (attributeBucket) {
+                transformedExecution = transformedExecution.withBuckets(
+                    definition.buckets[0],
+                    bucketSetTotals(definition.buckets[1], desiredRowTotals),
+                    bucketSetTotals(definition.buckets[2], desiredRowTotals),
```

Comment:
is this correct? Setting desiredRowTotals to both?

Reply:
Good catch!